flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/plan/rules/logical/PushProjectIntoTableSourceScanRuleTest.java

```diff
@@ -70,41 +69,40 @@ public void setup() {
 						" 'bounded' = 'true'\n" +
 						")";
 		util().tableEnv().executeSql(ddl2);
-	}
-
-	@Override
-	public void testNestedProject() {
-		expectedException().expect(TableException.class);
-		expectedException().expectMessage("Nested projection push down is unsupported now.");
-		testNestedProject(true);
-	}
 
-	@Test
-	public void testNestedProjectDisabled() {
-		testNestedProject(false);
-	}
-
-	private void testNestedProject(boolean nestedProjectionSupported) {
-		String ddl =
+		String ddl3 =
 				"CREATE TABLE NestedTable (\n" +
 						"  id int,\n" +
-						"  deepNested row<nested1 row<name string, `value` int>, nested2 row<num int, flag boolean>>,\n" +
-						"  nested row<name string, `value` int>,\n" +
+						"  deepNested row<nested1 row<name string, `value` int>, `nested2.` row<num int, flag boolean>>,\n" +
+						"  nested row<name string, `value.` int>,\n" +
 						"  name string\n" +
 						") WITH (\n" +
 						" 'connector' = 'values',\n" +
-						" 'nested-projection-supported' = '" + nestedProjectionSupported + "',\n" +
+						" 'nested-projection-supported' = 'true'," +
 						"  'bounded' = 'true'\n" +
 						")";
-		util().tableEnv().executeSql(ddl);
+		util().tableEnv().executeSql(ddl3);
+	}
 
+	@Override
+	@Test
+	public void testNestedProject() {
 		String sqlQuery = "SELECT id,\n" +
 				"    deepNested.nested1.name AS nestedName,\n" +
-				"    nested.`value` AS nestedValue,\n" +
-				"    deepNested.nested2.flag AS nestedFlag,\n" +
-				"    deepNested.nested2.num AS nestedNum\n" +
+				"    nested.`value.` AS nestedValue,\n" +
+				"    deepNested.`nested2.`.flag AS nestedFlag,\n" +
```

Comment:
how about keep the original test, and add a new test case the verify the case which field name contain dot